pyairtable/orm/model.py

```diff
@@ -170,6 +172,29 @@ def _validate_class(cls) -> None:
                 )
             )
 
+    def modify_kwargs(func):
+        # Modifies kwargs passed to Model.all and Model.first
+        # If Meta.use_field_ids, kwargs['return_fields_by_field_id']=True.
+
+        @wraps(func)
+        def wrapper(cls, *args, **kwargs):
+            kwargs = cls._kwargs_all_first(kwargs)
+            return func(cls, *args, **kwargs)
+
+        return wrapper
+
+    @classmethod
+    def _kwargs_all_first(cls, request_args: dict):
+        # Called by modify_kwargs to modify kwargs passed to `all()` & `first()`.
+        disallowed_args = (
+            "cell_format",
+            "return_fields_by_field_id",
+        )
+        args_union = {k: v for k, v in request_args.items() if k not in disallowed_args}
+        if cls._get_meta("use_field_ids"):
+            args_union["return_fields_by_field_id"] = True
+        return args_union
```

Comment:
If we follow the pattern above, I think we could probably simplify this down to something like:

```
kwargs = {
    "cell_format": "json",
    "user_locale": None,
    "time_zone": None,
    "return_fields_by_field_id": cls._get_meta("use_field_ids"),
}
```

I also wonder if this would be a good place to put `kwargs["fields"] = [...]` so that we don't retrieve more data from the API than our ORM actually requires.

Reply:
I thought we were going to outright remove the ability to pass "return_fields_by_field_id" without sanitizing the kwarg keys it allows those keys through.

I think we might have to then change some tests, I am getting:
`FAILED tests/test_orm.py::test_undeclared_field[first] - TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
FAILED tests/test_orm.py::test_undeclared_field[all] - TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'`

Reply:
Unless I change test_orm_model.py:
```
@pytest.mark.parametrize("methodname", ("all", "first"))
def test_passthrough(methodname):
    """
    Test that .all() and .first() pass through whatever they get.
    """
    with mock.patch(f"pyairtable.Table.{methodname}") as mock_endpoint:
        method = getattr(FakeModel, methodname)
        method(a=1, b=2, c=3)

    mock_endpoint.assert_called_once_with(a=1, b=2, c=3, cell_format="json", return_fields_by_field_id=False)
```

Reply:
That change looks reasonable to me. Where an Airtable API parameter has one obvious "right answer", it's probably better for orm.Model to be explicit about what it expects.